src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -332,41 +271,13 @@ default ClassDesc componentType() {
      * default package, or this {@linkplain ClassDesc} does not describe a class or interface type
      */
     default String packageName() {
-        if (!isClassOrInterface())
-            return "";
-        String desc = descriptorString();
-        int index = desc.lastIndexOf('/');
-        return (index == -1) ? "" : internalToBinary(desc.substring(1, index));
+        return "";
     }
 
     /**
-     * Returns a human-readable name for the type described by this descriptor.
-     *
-     * @implSpec
-     * <p>The default implementation returns the simple name
-     * (e.g., {@code int}) for primitive types, the unqualified class name
-     * for class or interface types, or the display name of the component type
-     * suffixed with the appropriate number of {@code []} pairs for array types.
-     *
-     * @return the human-readable name
+     * {@return a human-readable name for this {@code ClassDesc}}
```

Comment:
I don't see the merit of making the string representation unspecified.   The implSpec already specifies the string representation for the display name.   Why not just promoting it to normal spec text?

Reply:
I believe the specification for class or interface display name is too tight: if we have `java.awt.List` versus `java.util.List`, our current implementation prints `List` for both cases. It makes sense for an implementation to print something like `j.a.List` versus `j.u.List`.

Reply:
Printing the fully-qualified class name could also be an option.   Since the implementation is not changed, I would suggest promoting the implSpec text to normal spec text.   For a proposal to change the specification as well as the implementation to return something different with incompatibility, it'd good to do it as a separate RFE and discuss what was considered when the implSpec was decided to print the unqualified class name.

Reply:
I have promoted the implementation specification to API specification, describing that we will investigate in another RFE. Please review the latest revision and the CSR.